Microsoft.Toolkit.Uwp.UI/Triggers/IsEqualStateTrigger.cs

```diff
@@ -92,14 +93,21 @@ private static bool ConvertTypeEquals(object value1, object value2)
 
         private static object ConvertToEnum(Type enumType, object value)
         {
-            try
+            // value cannot be the same type of enum now
+            if (value is string str)
             {
-                return Enum.IsDefined(enumType, value) ? Enum.ToObject(enumType, value) : null;
+                return Enum.TryParse(enumType, str, out var e) ? e : null;
             }
-            catch
+
+            if (value is int || value is uint
+                || value is byte || value is sbyte
+                || value is long || value is ulong
+                || value is short || value is ushort)
```

Comment:
If you really want to use pattern matching to simplify the syntax, might as well go all the way 😄

```suggestion
            if (value is int or uint or byte or sbyte or long or ulong or short or ushort)
```

Reply:
Well, the project uses C# 8.0, and this requires C# 9.0. Is it generally OK to increase language version in a minor change?

Reply:
Yeah, we've been bumping the C# version in all projects lately. These are all purely syntactic sugar features, so except for very specific things (except eg. function pointers in some cases on UWP) they don't cause any issues. For instance, the base package and the `.Mvvm`, `.HighPerformance`, `.Diagnostics`, `.Uwp.UI.Animations` and `.Uwp.UI.Media` packages all use C# 9.0 already. Feel free to bump the language version in here and get access to the fancy new syntax 🙂